kvm-ioctls: backport kvm nested state to v0.22.1

phip1611 opened the pull request:
This is a backport of #322 [0] from the latest state to v0.22.1 to enable an easy way to use this in cloud-hypervisor by simply bumping the patch version of the component.

The backport is done in two steps, this is step 2/2. This commit only does the necessary steps for kvm-ioctls.

[0] https://github.com/rust-vmm/kvm/pull/322

### Summary of the PR

*Please summarize here why the changes in this PR are needed.*

### Requirements

Before submitting your PR, please make sure you addressed the following
requirements:

- [ ] All commits in this PR have Signed-Off-By trailers (with
  `git commit -s`), and the commit message has max 60 characters for the
  summary and max 75 characters for each description line.
- [ ] All added/changed functionality has a corresponding unit/integration
  test.
- [ ] All added/changed public-facing functionality has entries in the "Upcoming 
  Release" section of CHANGELOG.md (if no such section exists, please create one).
- [ ] Any newly added `unsafe` code is properly documented.

phip1611 committed:
kvm-ioctls: backport kvm nested state to v0.22.1

This is a backport of #322 [0] from the latest state to
v0.22.1 to enable an easy way to use this in cloud-hypervisor
by simply bumping the patch version of the component.

The backport is done in two steps, this is step 2/2. This
commit only does the necessary steps for kvm-ioctls.

[0] https://github.com/rust-vmm/kvm/pull/322

Signed-off-by: Philipp Schuster <philipp.schuster@cyberus-technology.de>
On-behalf-of: SAP philipp.schuster@sap.com

roypat commented:
heh, something weird is going on with the serde feature in CI, could you look into that? :o

phip1611 commented:
I have absolutely no clue what is going on there... I assume some Rust edition 2015 glitch. I'm looking for a workaround

snue commented:
> heh, something weird is going on with the serde feature in CI, could you look into that? :o

This is failing for the `kvm-bindings` build of the workspace, because the explicitly requested features do not have `fam-wrappers`, only `serde`. This was not an issue when the `kvm-ioctls` build requested the feature from the `kvm-bindings` crate of the workspace, but now it consumes directly from crates.io, and nothing forces the feature on the workspace build. You can see it uses both versions (0.12.1 and 0.12.0 from workdir) in the log.

https://buildkite.com/rust-vmm/kvm-ci/builds/1463/steps/canvas?jid=01995738-26ed-4e8c-9e14-c3f23d6de434

phip1611 committed:
ci: fix kvm-bindings

This is a back-port of [0] to fix the CI.

[0] https://github.com/rust-vmm/kvm/pull/351

Signed-off-by: Philipp Schuster <philipp.schuster@cyberus-technology.de>
On-behalf-of: SAP philipp.schuster@sap.com
